### PR TITLE
406 fe changes for pricing

### LINKED ISSUE #406: fe changes for pricing

### PR DESCRIPTION
**Pull request by roxy-dao**
Closes #406 

more fe changes for pricing... with this pricing should be complete now 😄 🤞

### CODE REVIEW

File: server/service/src/invoice_line/inbound_shipment_line/update/generate.rs [hunk 1]
```diff
@@ -85,23 +85,21 @@ fn generate_line(
...
     } else {
-        update_line.total_before_tax =
-            update_line.cost_price_per_pack * update_line.number_of_packs as f64;
-    }
-
-    if let Some(tax) = tax {
-        update_line.tax = tax.percentage;
-    }
+        update_line.cost_price_per_pack * update_line.number_of_packs as f64
+    };
```
Comment (clemens-msupply) [inline server/service/src/invoice_line/inbound_shipment_line/update/generate.rs:102]: why the `if` here? just use the else?

nit: use same cost_price_per_pack/number_of_packs order as in the if statement

File: server/service/src/invoice_line/outbound_shipment_line/update/generate.rs [hunk 2]
```diff
@@ -123,11 +123,17 @@ fn generate_line(
...
+        total_before_tax
+    } else if let Some(number_of_packs) = input.number_of_packs {
+        update_line.sell_price_per_pack * number_of_packs as f64
+    } else if input.stock_line_id.is_some() || input.item_id.is_some() {
+        sell_price_per_pack * number_of_packs as f64
     } else {
-        update_line.total_before_tax = sell_price_per_pack * number_of_packs as f64;
-    }
+        sell_price_per_pack * number_of_packs as f64
+    };
```
Comment (clemens-msupply) [inline server/service/src/invoice_line/outbound_shipment_line/update/generate.rs:135]: The two if else and the last else block all do the same, right? maybe just use the last else?

**Review by clemens-msupply — APPROVED** [review]
Looks good to me, just some comment about IMHO redundant if else block. Not sure if this was intended? if yes maybe add a comment why.

File: server/service/src/invoice_line/outbound_shipment_line/update/generate.rs (hunk 2 above)
Comment (roxy-dao) [reply server/service/src/invoice_line/outbound_shipment_line/update/generate.rs:135]: 🤔  That's what I thought... [but was told to add it in](https://github.com/openmsupply/open-msupply/pull/306#pullrequestreview-1042585516)

Comment (clemens-msupply) [reply server/service/src/invoice_line/outbound_shipment_line/update/generate.rs:135]: I might be wrong but think @andreievg just wanted to be very clear about the logic but not how exactly it is implemented (?). Find the ifs in combination with the last else confusing, maybe put it in a comment when the total after tax should be recalculated and why do the last else.

Comment (roxy-dao) [reply server/service/src/invoice_line/outbound_shipment_line/update/generate.rs]: The last else was supposed to be total before tax 🤔

Comment (andreievg) [reply server/service/src/invoice_line/outbound_shipment_line/update/generate.rs]: The logic looks sound, but yeah a comment would be great

Comment (andreievg) [reply server/service/src/invoice_line/outbound_shipment_line/update/generate.rs]: Ohh sorry, it looks sounds now, i just saw a commit you did

**Pull request merged**